provisioner/provision.js

```diff
@@ -79,61 +80,90 @@ var stats = {
   failure: 0,
 };
 
+// For when you want to be really certain that the program will
+// exit
+function exitTimer(time) {
+  var t = time || 30000;
+  setTimeout(() => {
+    debug('hey, so you probably are trying to figure out');
+    debug('why this process suddenly disappeared.  a major');
+    debug('error occured and you only get %d ms to exit after', t);
+    debug('before we process.exit(44). handle things faster!');
+    process.exit(44);
+  }, t);
+}
+
 /**
  * Start running a provisioner.
  */
 Provisioner.prototype.run = function () {
-  var that = this;
-
   this.__keepRunning = true;
+
   this.__watchDog.on('expired', function () {
-    debug('[alert-operator] Provisioner is stuck, killing');
+    debug('[alert-operator] provisioning iteration exceeded max time');
+    exitTimer(MAX_KILL_TIME);
     throw new Error('WatchDog expired');
```

Comment:
Oh, yeah I agree with that :)
I would just rely on `process.exit` to avoid having to even think about the other flow... not that it can do any harm :)